File: Fbs.WebApi/Entities/Facility.cs
using Fbs.WebApi.Repository;

namespace Fbs.WebApi.Entities;

public class Facility : Entity
{
    public string? Name { get; set; }
    public string? Group { get; set; }
    public List<string>? Scope { get; set; }

    public override string? GetId() => Name;
    public bool AvailableForAll => Scope?.Contains("All") ?? false;
}

File: Fbs.WebApi/src/main/kotlin/sg/from/fbs/entities/Facility.kt
package sg.from.fbs.entities

data class Facility(
    var row: Int = 0,
    var name: String? = null,
    var group: String? = null,
    var scope: List<String>? = null
) {
    val availableForAll: Boolean
        get() = scope?.contains("All") == true
}